### PR TITLE
Simplify custom schedulers - OOP is BAD ;)

### PR DESCRIPTION
**Pull request by langara**

### CODE REVIEW

File: src/main/kotlin/com/elpassion/intelijidea/common/ProgressScheduler.kt [hunk 1]
```diff
@@ -4,17 +4,12 @@ import com.intellij.openapi.progress.ProgressIndicator
...
 
-class ProgressScheduler(private val project: Project?, private val description: String) : Scheduler() {
-    override fun createWorker(): Worker {
-        return ExecutorScheduler.ExecutorWorker { command ->
-            ProgressManager.getInstance().run(object : Task.Backgroundable(project, description) {
-                override fun run(indicator: ProgressIndicator) {
-                    command.run()
-                }
-            })
+fun ProgressScheduler(project: Project?, description: String) = ExecutorScheduler { command ->
```
Comment (pawegio) [inline src/main/kotlin/com/elpassion/intelijidea/common/ProgressScheduler.kt]: that's why in Kotlin there is no `new` keyword 😃

File: src/main/kotlin/com/elpassion/intelijidea/task/UINonModalScheduler.kt [hunk 2]
```diff
@@ -2,19 +2,8 @@ package com.elpassion.intelijidea.task
...
-    override fun createWorker(): Worker {
-        return ExecutorScheduler.ExecutorWorker(UIExecutor)
-    }
-
-    private object UIExecutor : Executor {
-        override fun execute(command: Runnable) {
-            ApplicationManager.getApplication().invokeLater(command, ModalityState.NON_MODAL)
-        }
-    }
+val UINonModalScheduler = ExecutorScheduler { command ->
```
Comment (pawegio) [inline src/main/kotlin/com/elpassion/intelijidea/task/UINonModalScheduler.kt]: I would now put both properties in a single kotlin file

File: src/main/kotlin/com/elpassion/intelijidea/common/ProgressScheduler.kt (hunk 1 above)
Comment (OrdonTeam) [inline src/main/kotlin/com/elpassion/intelijidea/common/ProgressScheduler.kt]: Mind bolwn

**Comment by jozzi05** [comment]
LGTM

**Pull request merged**